code/game/machinery/doors/windowdoor.dm

```diff
@@ -590,3 +590,41 @@
 	dir = SOUTH
 	icon_state = "rightsecure"
 	base_state = "rightsecure"
+
+/obj/machinery/door/window/checkpoint
+	icon_state = "sec_left"
+	base_state = "sec_left"
+	layer = ABOVE_MOB_LAYER
+	closingLayer = ABOVE_MOB_LAYER
```

Comment:
why does the layer need to be specified here? does the parent type not provide a fitting layer?

Reply:
Heya, I did this part.

Rails are set to a lawyer below mobs, which means that the mob will always appear on top.

Since this is used in such a small place that effect becomes very apparent and the illusion of a checkpoint is ruined which would negatively impact immersion.

I've decided to give them the above mob layer to account for this.
The only way this would be an issue is with a mob bigger than one tile (like a big moth with growth serum) but at that point the immersion is ruined anyway.